lib/model.d.ts

```diff
@@ -1840,14 +1851,18 @@ export abstract class Model {
    * always be called with a single instance.
    */
   static findById<M extends Model>(this: {new (): M} & typeof Model, identifier?: number | string, options?: FindOptions): Promise<M | null>;
+  static findById<M extends Model>(this: {new (): M} & typeof Model, identifier: number | string, options: NonNullFindOptions): Promise<M>;
```

Comment:
What about other options? What about passing `false`? I think it would be better to handle this overload with an intersection type.

Reply:
True, in our code we only passed ` true` to it hence the oversight, I'll look into making adjustments.

Reply:
TS doesn't seem to differentiate between `FindOptions` with the added property `rejectOnEmpty?: boolean` eg.

`  static findOne<M extends Model>(this: {new (): M} & typeof Model, options: FindOptions & { rejectOnEmpty: true }): Promise<M>;
`

will not be preferred over the less specific `rejectOnEmpty: boolean` inside `FindOptions` therefor the overload doesn't work. 

A solution would be to create 2 interfaces inheriting  from `FindOptions` but that seems rather hacky.

Another solution looks like this:

```
static findOne<M extends Model>(this: {new (): M} & typeof Model, options?: FindOptions & { rejectOnEmpty?: false }): Promise<M | null>;
static findOne<M extends Model>(this: {new (): M} & typeof Model, options: FindOptions & { rejectOnEmpty: true }): Promise<M>;
```

this one _works_ but is very verbose, which is kind of annoying considering these should also (technically) need to be added to 1:1 relation getters.

Can you think of a better solution?